### PR TITLE
Cyrillux20240530

### PR DESCRIPTION
**Pull request by nvolk**
- New Cyrillux validator (move this functionality away from UI)

- also tune old usemarcon-cyrillux related rules

### CODE REVIEW

File: src/cyrillux.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      ...field.subfields,
+      {code: '9', value: 'CYRILLIC <TRANS>'}
+    ];
+
+    field.subfields = subfields; // eslint-disable-line functional/immutable-data
+    field.tag = '880'; // eslint-disable-line functional/immutable-data
+    return field;
+  }
+
+  function convertFieldToSfs4900Field880(field, occurrenceNumber) {
```
Comment (natlibfi-jonollil) [inline src/cyrillux.js:158]: Could it work like this?
```js
function convertFieldToSfs4900Field880(field, occurrenceNumber) {
  // Looks like there's no need to fix composition, but I haven't tested extensively/checked
  return {
    tag: '880',
    ind1: field.ind1,
    ind2: field.ind2,
    subfields: [
      {code: '6', value: `${field.tag}-${occurrenceNumber}`},
      ...field.subfields.map(sf => convertSubfieldToSfs4900(sf)),
      {code: '9', value: 'SFS4900 <TRANS>'}
    ]
  };
}
```

File: src/cyrillux.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    return res;
+  }
+
+  function validateField(field, res, record) {
+    const orig = fieldToString(field);
+
+    const normalizedFields = processField(clone(field), record);
+    const mod = fieldsToString(normalizedFields).replace(/\t__SEPARATOR__\t/ug, ', ').replace(/ (‡6 [0-9][0-9][0-9])-[0-9][0-9]+/gu, ' $1-NN'); // eslint-disable-line prefer-named-capture-group
+    if (orig !== mod) { // Fail as the input is "broken"/"crap"/sumthing
+      res.message.push(`TODO: ${orig} => ${mod}`); // eslint-disable-line functional/immutable-data
```
Comment (natlibfi-jonollil) [inline src/cyrillux.js:74]: todo?

**Review by natlibfi-jonollil — CHANGES_REQUESTED** [review]

File: src/cyrillux.js (hunk 2 above)
Comment (nvolk) [reply src/cyrillux.js:74]: Joo, TODO on huono. Toi siis kertoo tehtävän muutokset. Vaihdan tohon termin...

File: src/cyrillux.js (hunk 1 above)
Comment (nvolk) [reply src/cyrillux.js]: Yes.

**Pull request merged**